File: Project/C++/Project1/Buyer.cpp
#include "Buyer.h"

//Buyer::Buyer(){}
Buyer::Buyer(std::string name, std::string email, std::string address, int postalCode){
	setName(name);
	setAddress(address);
	setEmail(email);
	setPostalCode(postalCode);
	buyerCategory = BRONZE;
	cart = ShoppingCart(buyerCategory);
	cart.setBuyerCategory(buyerCategory);

}

int Buyer::getBuyerCategory()
{
	return buyerCategory;
}

void Buyer::setBuyerCategory()
{
	if (buyerPts >= 100) {
		buyerCategory = SILVER;
	}
	else if (buyerPts >= 200) {
		buyerCategory = GOLD;
	}
	cart.setBuyerCategory(buyerCategory);
}

int Buyer::getBuyerPts()
{
	return buyerPts;
}

void Buyer::setBuyerPts(int buyerPts)
{
	this->buyerPts = buyerPts;
}

void Buyer::awardBonus(int pts)
{
	setBuyerPts(buyerPts + pts);
	setBuyerCategory();
}

void Buyer::placeOrder(Item* item, int quantity)
{
	cart.addItemOrdered(item, quantity);
}

File: Project/Java/src/eshop/Buyer.java
package eshop;

public class Buyer extends User {
    public static final int BRONZE = 0;
    public static final int SILVER = 1;
    public static final int GOLD = 2;
    private int buyerPts = 0;
    private int buyerCategory;
    public ShopingCart cart = new ShopingCart(buyerCategory);

    Buyer(String name, String email, String address, int postalCode){
        setName(name);
        setEmail(email);
        setAddress(address);
        setPostalCode(postalCode);
        this.buyerCategory = BRONZE;
    }

    public int getbuyerCategory() {
        return buyerCategory;
    }

    public void setbuyerCategory() {
        if(buyerPts>=100){
            buyerCategory = SILVER;
        }else if(buyerPts>=200){
            buyerCategory = GOLD;
        }
        cart.setBuyerCategory(buyerCategory);
    }

    public int getBuyerPts() {
        return buyerPts;
    }

    public void awardBonus(double price){
        buyerPts+=(int)(price*0.1);
        setbuyerCategory();
    }
    public void placeOrder(Item item, int quantity){
        try {
            ShopingCart.addItemOrdered(item, quantity);
        } catch (ItemOutOfStockException e) {
            System.out.println("The item is out of stock");
        }
    }
}
